README.md

```diff
@@ -0,0 +1,172 @@
+# Common interfaces for vision models.
+
+This package provides common interfaces for vision models. 
+
+Each Model must have two additional torch.nn.Module; criterion and predictor. A criterion is used in the training phase to calculate the loss from the model outputs. A predictor is used in test phase to get final prediction results from the model outputs. A Model must define criterion() and predictor() property to return a corresponding module. The expected input/output formats for model, criterion, predictor are defined for each task.
+
+When a model requires advanced changes to training behavior, it can implement a Plugin class inheriting PluginBase. For the detail, please see the description of the PluginBase class.
+
+## Install
+```
+pip install vision-model
+```
+
+## Examples
+
+Here is an example to define a simple multiclass classification model.
+```python
+from vision_model import ModelBase, PluginBase
+
+class MyModel(ModelBase):
+    def __init__(self):
+        super().__init__()
+        self.model = torch.nn.Sequence([torch.nn.Conv2d(...)])
+    
+    def forward(self, x):
+        return self.model(x)
+
+    def change_something(self):
+        pass  # Do something here if needed.
+
+    @property
+    def plugins(self):
+        return [MyPlugin()]
+
+    @property
+    def criterion(self):
+        return torch.nn.CrossEntropyLoss()
+
+    @property
+    def predictor(self):
+        return torch.nn.Softmax(1)
+
+
+class MyPlugin(PluginBase):
+    def on_train_start(self, trainer, model):
+        # It can make some changes to the model when a training begins.
+        # See help(PluginBase) to find available callback methods.
+        model.change_something()
+        return
+```
+
+Pleasee see examples/ directory for more practical examples.
+
+
+## Interfaces
+
+```
```

Comment:
seems redundant with the real code

Reply:
some people are too lazy to go into source codes, including me lol This is auto-generated. Makes it easy to get the whole picture.

Reply:
maybe add a link to the file could help

Reply:
ok. removed from README. maybe I'll make a github page by pdoc and write a link on README.